addons/zeus/functions/fnc_moduleStaticWeaponPlaceLocal.sqf

```diff
@@ -0,0 +1,91 @@
+/*
+ * Author: PabstMirror
+ * Adapted from BIS_fnc_packStaticWeapon (Dean "Rocket" Hall, reworked by Killzone_Kid)
+ * Place on a unit in a static weapon to make them dissasemble and pickup the backpacks parts
+ *
+ * Arguments:
+ * 0: Gunner (carrying the weapon backpack) <OBJECT>
+ * 1: Watch Direction <ARRAY>
+ * 2: Module Placer Client ID <NUMBER>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [q1, [0,0,0], 2] call ace_zeus_fnc_moduleStaticWeaponPlaceLocal
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_gunner", "_watchPosASL", "_callerID"];
+TRACE_3("moduleStaticWeaponPlaceLocal",_gunner,_watchPosASL,_callerID);
+
+if ((!alive _gunner) || {!local _gunner}) exitWith {};
+private _cfgBase = configFile >> "CfgVehicles" >> (backpack _gunner) >> "assembleInfo" >> "base";
+private _compatibleBases = if (isText _cfgBase) then {[getText _cfgBase]} else {getArray _cfgBase};
+
+private _availableUnits = (units group _gunner) select {(!isNull _x) && {_gunner != _x} && {[_x] call EFUNC(common,isAwake)} && {(vehicle _x) == _x}};
+private _assistant = objNull;
+{
+    private _unit = _x;
+    if (({(unitBackpack _unit) isKindOf _x} count _compatibleBases) > 0) exitWith {_assistant = _x};
+} forEach _availableUnits;
+TRACE_2("",_gunner,_assistant);
+if (!local _assistant) exitWith {WARNING("Split locality");};
```

Comment:
You already calculate the base/assistant in the calling function, should those just be passed in, instead of recalculating it?